multidim-interop/js/v0.41/test/ping.spec.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-console */
+/* eslint-env mocha */
+
+import { } from 'aegir/chai'
+// import { createClient } from 'redis'
+
+import { createLibp2p, Libp2pOptions } from 'libp2p'
+// import { tcp } from '@libp2p/tcp'
+import { webSockets } from '@libp2p/websockets'
+import { noise } from '@chainsafe/libp2p-noise'
+import { mplex } from '@libp2p/mplex'
+import { yamux } from '@chainsafe/libp2p-yamux'
+import { multiaddr } from '@multiformats/multiaddr'
+
+describe('ping test', () => {
+  it('should ping', async () => {
+    let nodeFetchImport = "node-fetch"
+    let fetch = (typeof window === "undefined") ?
+      (await import(nodeFetchImport)).default :
+      window.fetch
+
+    const TRANSPORT = process.env.transport
+    const SECURE_CHANNEL = process.env.security
+    const MUXER = process.env.muxer
+    const IS_DIALER_STR = process.env.is_dialer
+    const IP = process.env.ip
+
+    const isDialer = IS_DIALER_STR === 'true'
+    const options: Libp2pOptions = {
+      start: true
+    }
+
+    switch (TRANSPORT) {
```

Comment:
Cleaner I think might be to turn this into an if else and disable tcp case if browser. And the regular import is also cleaner I think with the package.json to just turn @libp2p/tcp to false (so an empty import), this keeps all imports at top.

Reply:
Is this supported by all bundlers? Is that part of the package.json spec? https://docs.npmjs.com/cli/v9/configuring-npm/package-json#browser doesn't say anything about that. https://github.com/defunctzombie/package-browser-field-spec#ignore-a-module This does, but I'm not sure if that's well supported.

Reply:
Fairly certain it should be yes. I always use webpack, def works for that. I imagine stuff like Vite as well. I mean it is really made for bundlers so it would be odd if a bundler does not support it. It's also pretty nice as i works for any dependency as well, no matter how deep in the chain.

Reply:
The `browser` field isn't documented by npm as it's a webpack convention that was adopted by other bundlers - the defunctzombie spec linked to above is probably as good as we're ever going to get..

Reply:
In that case, I feel like it’s better to do the dynamic import here which is part of the ES spec.

Reply:
Using the regular `import` keyword and the `browser` field to mark the import `false` usually has fewer hidden surprises as the keyword is static code which is easily analysed by bundlers - dynamic imports sometimes cause problems as they are evaluated at runtime by which point the bundler isn't involved any more.

Reply:
Even if it wouldn't, dynamic imports like this are a bit dirty, no? Cleaner to see them all listed on top, of course one could argue that indirect dependencies are also not visible, but dunno if one should go there.

I have either way also no strong opinion, but it does make sense to me that bundlers do not take this into account, as that would mean to support it for real they would need to pretty much run the entire javascript code to know for sure what is required and what not.